ledger/test-common/src/main/daml/model/Test.daml

```diff
@@ -613,3 +613,34 @@ template Asset
       AssetTransfer: ContractId Asset
         with newOwner: Party
         do create this with owner = newOwner
+
+template MultiPartyContract
```

Comment:
Together with the actual tests above, this covers the common uses of multi-party submissions, but only at the command level. `actAs` only has an effect at the command level as it sets the available authorizers for top-level actions, but `read` as has an effect on any action in the transaction tree.

For completeness, I would therefore add a few tests that test reads at least one level down in the transaction tree. For example:

1. Give the template a contract key `key this: MultiPartyContract; maintainer key.parties`.
2. Give the template three choices `FetchOther, FetchOtherByKey, LookupOther`, which take a cid, or key or a second contract instance, respectively.
3. Test all 12 combinations of
  - 3 choices
  - Well-authorized or not
  - Contract to be looked up visible or not

Reply:
Is the following an example of how this should work? 

1. Alice creates a `MultiPartyContract` for herself (let's call it `cA`)
2. Bob creates a `MultiPartyContract` for himself (let's call it `cB`)
3. Bob exercise a `FetchOther` choice on his contract `cB`, which attempts to fetch contract `cA`. This is well-authorized, but the contract lookup will fail unless Alice is a `readAs` party or `cA` has been divulged to Bob.

Reply:
https://github.com/digital-asset/daml/blob/70af09db51e9f0d2b5733f108e06e096526cf02f/compiler/damlc/tests/src/DA/Test/ScriptService.hs#L653 might be a reasonable example from the daml script tests.

Reply:
@kamil-da This seems to be resolved, isn't it?

Reply:
Yes, @rautenrieth-da created additional tests. (See: https://github.com/digital-asset/daml/pull/8266/commits/acd3745c6fc14d66fe3200d2b3b5fd660facd3d6)